src/main/python/pgshovel/streams/utilities.py

```diff
@@ -24,3 +32,33 @@ def __str__(self):
             active=','.join(map(str, active[:self.max])),
             truncated='' if len(active) <= self.max else ',+{0}...'.format(len(active) - self.max),
         )
+
+
+def is_start_batch_operation(message):
+    """
+    Is the ``message`` the start operation of a batch of mutations.
+    """
+    value = get_oneof_value(message.batch_operation, 'operation')
+    return type(value) in TRANSACTION_START_EVENT_TYPES
+
+
+def prime_until(stream, check, max_messages):
+    """
+    Attempts to "prime" the ``stream`` by dropping any messages for which
+    ``check`` evaluates to ``False``, up until ``max_messages``.
+    """
+
+    attempts = count(start=1)
+
+    def still_checking((offset, message)):
+        attempt = next(attempts)
```

Comment:
The first attempt here is `2` which is a little confusing at first read.

(Also not sure that I'd assign it to a local, but that's just my stylistic preference.)